src/js/client/editprefset-dialogs-widget.js

```diff
@@ -0,0 +1,348 @@
+/**
+ * Dialog boxes
+ *
+ * - addContextDialog Adds a preference set to a preferences safe
+ * - editContextDialog Edits metadata for a preference set
+ * - confirmDeleteContextDialog Confirms with the user before permanently
+ *   removing a preference set from a safe
+ * - confirmAddProductDialog After clicking a product to add to
+ *   preference safe, confirms before actual insertion.
+ * - confirmSaveDialog Shows a list of changes since last save, and
+ *   confirms with the user before saving preference set.
+ * - confirmRemoveProductDialog Confirms with the user before permanently
+ *   removing a product from the preference safe.
+ *
+ * Copyright 2019 Raising the Floor - International
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ *
+ * You may obtain a copy of the License at
+ * https://github.com/GPII/universal/blob/master/LICENSE.txt
+ */
+"use strict";
+
+var gpii = fluid.registerNamespace("gpii");
+fluid.registerNamespace("gpii.devpmt");
+fluid.registerNamespace("gpii.devpmt.dialogs");
+
+/**
+ * Add Preferences Set Context Dialog
+ */
+fluid.defaults("gpii.devpmt.dialogs.addContextDialog", {
+    gradeNames: ["gpii.devpmt.dialogs.confirmDialog"],
+    templates: {
+        initial: "editprefset-addContext-dialog"
+    },
+    model: {
+        contextId: "",
+        contextToCopy: "",
+        contextNames: [],
+        fieldErrors: ""
+    },
+    selectors: {
+        contextIdInput: "#pmt-add-context-name-input",
+        contextToCopySelect: "#pmt-context-to-copy-select"
+    },
+    bindings: {
+        contextIdInput: "contextId",
+        contextToCopySelect: "contextToCopy"
+    },
+    invokers: {
+        acceptConfirmDialog: {
+            funcName: "gpii.devpmt.dialogs.addContextDialog.acceptConfirmDialog",
+            args: ["{that}.model", "{that}.applier", "{that}.closeDialog", "{arguments}.0"]
+        }
+    },
+    modelListeners: {
+        "fieldErrors": {
+            func: "{that}.reRender"
+        }
+    }
+});
+
+/**
+ * Dialog for adding a new preference set (context) to a preferences safe (prefset).
+ * Includes a dropdown that can be used to select a context to copy the
+ * initial settings from.
+ *
+ * @param {Object} model - Model object
+ * @param {changeApplier} applier - Change applier for model
+ * @param {Function} closeDialog - Zero-arg function to close dialog
+ * @param {event} event - Browser event triggered by the accept button.
```

Comment:
DOMEvent or so is probably clearer.

Reply:
great, thanks, replaced all the {event}'s